docs/interview/JavaScript/02-medium/02-medium.md

```diff
@@ -1 +1,42 @@
 ### JavaScript Medium
+
+<details>
+<summary>What Object destructuring in Javascript?</summary>
+
+Object destructuring is the special feature in JavaScript. It is a convenient way to extract data from JavaScript objects and assign them to the variable in more readable way. 
+
+Object destructuring allows you to write less code and make it more readable, especially when working with objects with many properties.
+
+Object destructuring provides shorthand syntax to extracting value from the objects and array.
+
+**Syntax :**
+
+```js
+const { } = object-name;
```

Comment:
rename `object-name` to `objectName`

in js, we follow camelCase for normal variables.

Rest looks good to me.